File: ChainOfResponsibility/NetworkUser.m
//
//  NetworkUser.m
//  ChainOfResponsibility
//
//  Created by Maksim Ivanov on 28/04/2019.
//  Copyright © 2019 Maksim Ivanov. All rights reserved.
//

#import "NetworkUser.h"
#import "TCPIPNetworkAccessLayer.h"
#import "TCPIPInternetworkLayer.h"
#import "TCPIPTransportLayer.h"
#import "TCPIPApplicationLayer.h"


@implementation NetworkUser

- (void)imitateNetworkRequest
{
    NSLog(@"NetworkUser imitateNetworkRequest");
    
    TCPIPNetworkAccessLayer *networkAccessLayer = [TCPIPNetworkAccessLayer new];
    
    TCPIPInternetworkLayer *internetworkLayer = [TCPIPInternetworkLayer new];
    internetworkLayer.nextTCPIPLayer = networkAccessLayer;
    
    TCPIPTransportLayer *transportLayer = [TCPIPTransportLayer new];
    transportLayer.nextTCPIPLayer = internetworkLayer;
    
    TCPIPApplicationLayer *applicationLayer = [TCPIPApplicationLayer new];
    applicationLayer.nextTCPIPLayer = transportLayer;
    
    NSString *applicationData = @"some application data";
    [applicationLayer makeRequest:[NSMutableArray arrayWithArray:@[applicationData]]];
}

@end

File: ChainOfResponsibility/NetworkUser.swift
//
//  NetworkUser.swift
//  ChainOfResponsibility
//
//  Created by Maksim Ivanov on 28/04/2019.
//  Copyright © 2019 Maksim Ivanov. All rights reserved.
//

class NetworkUser {
    
    func imitateNetworkRequest() {
        print("NetworkUser imitateNetworkRequest");
        
        let networkAccessLayer = TCPIPNetworkAccessLayer()
        
        let internetworkLayer = TCPIPInternetworkLayer()
        internetworkLayer.nextTCPIPLayer = networkAccessLayer
        
        let transportLayer = TCPIPTransportLayer()
        transportLayer.nextTCPIPLayer = internetworkLayer;
        
        let applicationLayer = TCPIPApplicationLayer()
        applicationLayer.nextTCPIPLayer = transportLayer
        
        let applicationData = "some application data";
        
        applicationLayer.makeRequest(data: [applicationData]);
    }
}
